fuse/utils/file_io/path.py

```diff
@@ -28,3 +28,26 @@ def get_extension(filepath: str) -> str:
         return ""
     ans = _basename[last_dot:]
     return ans
+
+
+def remove_extension(filepath: str) -> str:
+    """
+    Returns the filename without the extension
+    For example - remove_extension('/a/b/c/d/asdf.txt') with return '/a/b/c/d/asdf'
+    """
+
+    _dirname = os.path.dirname(filepath)
+    _basename = os.path.basename(filepath)
+    if '.' not in _basename:
+        return filepath
+
+    last_dot = _basename.rfind(".")
+    if last_dot < 0:
```

Comment:
It can't happen

Reply:
why not? it represents the case that there is no '.' at all in the filename (and therefore no extension to remove)
for example - /a/b/c/foo

-1 just marks that it didn't find it

In [1]: 'abcd'.rfind('z')
Out[1]: -1

Reply:
Just because you already checked it in line 41 and returned the original filepath in such a case.

Reply:
oh I see! so it's redundant, will fix - thanks!

Reply:
done